docs/spec/consensus/fork-accountability.md

```diff
@@ -0,0 +1,142 @@
+
+Example scenario: in a round r of height h we have 1/3 of correct validators precommitting a value A, 1/3 of correct
+validators precomit nil, 1/3 of faulty validators don't send any message and 1 faulty validator precommit nil.
+In some round r' > r, +1/3 of faulty validators together with 1/3 of correct that haven't locked any value commit some other value B. +1/3 of faulty validators now "go back to the past" and sign precomit message for value A in round r.
+Together with precomit messages of correct processes this for a commit for value A. Note that only a faulty validator
+that previously precomit nil did equivocation, while other 1/3 of faulty validators actually executed an
+attack that has exactly the same sequence of messages as part of amnesia attack. Detecting this kind of attack boil down to mechanisms for equivocation and amnesia. 
+
+Q: should we keep this as a separate kind of attacks? It seems that equivocation, amnesia and fantom validators 
+are the only kind of attack we need to support and this gives us security also in other cases. This would not
+be surprising as equivocation and amnesia are attacks that followed from the protocol and fantom attack is 
+not really an attack to Tendermint but more to the Proof of stake module. 
+
+### Fantom validators
+
+In case of fantom validators, processes that are not part of the current validator set but are still bonded (
+as attack happen during their unbonding period) can be part of the attack by signing vote messages. This attack 
+can be executed against both full nodes and lite clients. 
+
+Example scenario: a lite client has a trust in a header for height h (and the corresponding validator set VS1). 
+As part of bisection header verification it verifies header at height h+k with new validator set VS2. 
+Some of the validators from VS2 signed the header at height h+k that is different from the corresponding header
+at the main chain. Detecting these kind of attacks is easy as it just requires verifying if processes are signing messages in heights in which they are not part of the validator set.  
+
+Note that we can have fantom validators based attacks as a follow up of equivocation or amnesia based attack where 
+forked state contains validators that are not part of the validator set at the main chain. In this case they keep
+signing messages contributed to a forked chain although they are not part of the validator set on the main chain.
+This attack can also be used to attack full node during a period of time he is eclipsed. 
+
+### Lunatic validator
+
+Lunatic validator agrees to sign commit messages for arbitrary application state. It is used to attack lite clients. 
+Note that detecting this behavior require application knowledge. Detecting this behavior can probably be done by
+referring to the block before the one in which height happen.  
+
+Q: can we say that in this case a validator ignore to check if proposed value is valid before voting for it?
```

Comment:
I think so, yes and we should punish for it

Reply:
Agreed, I wonder if this is complex if combined with differing rounds, it seems like we might need some sort of challenge-response where a validator which signed a state that didn't end up being part of the canonical chain must submit the sequence of transactions which led to it in order to avoid being slashed.